### PR TITLE
feat(cketh): NNS proposals to upgrade ckETH ledger suite to add support for ICRC-21

### PR DESCRIPTION
**Pull request by gregorydemay**
([XC-148](https://dfinity.atlassian.net/browse/XC-148)): NNS proposals to upgrade the ckETH ledger suite (ledger, index and archive canisters) to `3d0b3f10417fc6708e8b5d844a0bac5e86f3e17d` to add support for the [ICRC-21: Canister Call Consent Messages](https://github.com/dfinity/wg-identity-authentication/blob/fd846030109710cab67d9381485a73db424f2b07/topics/ICRC-21/icrc_21_consent_msg.md) standard. 

[XC-148]: https://dfinity.atlassian.net/browse/XC-148?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: rs/ethereum/cketh/mainnet/cketh_archive_upgrade_2024_08_19.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+Upgrade the ckETH archive canister to the latest version to add support for the [ICRC-21: Canister Call Consent Messages](https://github.com/dfinity/wg-identity-authentication/blob/fd846030109710cab67d9381485a73db424f2b07/topics/ICRC-21/icrc_21_consent_msg.md) standard.
+
+
+## Upgrade args
+
+```
+git fetch
+git checkout 3d0b3f10417fc6708e8b5d844a0bac5e86f3e17d
+cd rs/rosetta-api/icrc1/archive
+didc encode '()'
```
Comment (gregorydemay) [inline rs/ethereum/cketh/mainnet/cketh_archive_upgrade_2024_08_19.md:23]: `4449444c0000`

File: rs/ethereum/cketh/mainnet/cketh_index_upgrade_2024_08_19.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+Upgrade the ckETH index canister to the latest version to add support for the [ICRC-21: Canister Call Consent Messages](https://github.com/dfinity/wg-identity-authentication/blob/fd846030109710cab67d9381485a73db424f2b07/topics/ICRC-21/icrc_21_consent_msg.md) standard.
+
+
+## Upgrade args
+
+```
+git fetch
+git checkout 3d0b3f10417fc6708e8b5d844a0bac5e86f3e17d
+cd rs/rosetta-api/icrc1/index-ng
+didc encode -d index-ng.did -t '(opt IndexArg)' '(null)'
```
Comment (gregorydemay) [inline rs/ethereum/cketh/mainnet/cketh_index_upgrade_2024_08_19.md:23]: `4449444c066e016b02fcb88b840302b0ced18403056c02f1f7fcf70603dcb79b830b046e686e786c02f1f7fcf70668dcb79b830b04010000`

File: rs/ethereum/cketh/mainnet/cketh_ledger_upgrade_2024_08_19.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+Upgrade the ckETH ledger canister to the latest version to add support for the [ICRC-21: Canister Call Consent Messages](https://github.com/dfinity/wg-identity-authentication/blob/fd846030109710cab67d9381485a73db424f2b07/topics/ICRC-21/icrc_21_consent_msg.md) standard.
+
+
+## Upgrade args
+
+```
+git fetch
+git checkout 3d0b3f10417fc6708e8b5d844a0bac5e86f3e17d
+cd rs/rosetta-api/icrc1/ledger
+didc encode -d ledger.did -t '(LedgerArg)' '(variant {Upgrade = null})'
```
Comment (gregorydemay) [inline rs/ethereum/cketh/mainnet/cketh_ledger_upgrade_2024_08_19.md:23]: `4449444c1e6b02fcb88b840301b0ced18403186e026c0a92cdb6f902039efeb9a40309f2c794ae030aefcee780040b97aabdbb06058484d5c0070585f199f40710b0d7c3920b1591c9aafe0d09bea3d1c30f166e046c089ea581d20105b2a7c2d20305a495a5e90605ffb08aab0806e0ab86ef0805e4d8cce80b0593c8e6c70c05dec5d8ae0e086e786e076d686e686e716e7d6e0c6d0d6c020071010e6b04cf89df017cc189ee017dfdd2c9df020fcdf1cbbe03716d7b6e116b029d83f46a12c9c5f1d0037f6c02b3b0dac30368ad86ca8305136e146d7b6e7a6e176c01c7bfe7b60b7e6c0dc295a99301199efeb9a40371f2c794ae037defcee780040caecbeb880412b2a4dab2051a97aabdbb06058484d5c0070582babe82081ca1e5f7a10a1db0d7c3920b1591c9aafe0d71bea3d1c30f166e7b6d1b6c020012017d6e126c089ea581d20178b2a7c2d20305a495a5e90678ffb08aab0806e0ab86ef0805e4d8cce80b0593c8e6c70c05dec5d8ae0e6801000000`

File: rs/ethereum/cketh/mainnet/cketh_archive_upgrade_2024_08_19.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
+# Proposal to upgrade the ckETH archive canister
```
Comment (gregorydemay) [inline rs/ethereum/cketh/mainnet/cketh_archive_upgrade_2024_08_19.md:1]:
```
~/bin/ic-admin \                                                                                                                                                                                                                                                    
    --use-hsm \
    --key-id 01 \
    --slot 0 \
    --pin $(cat ~/.hsm-pin) \
    --nns-url "https://ic0.app" \
    propose-to-change-nns-canister \
    --proposer 17212304975669116357 \
    --canister-id xob7s-iqaaa-aaaar-qacra-cai \
    --mode upgrade \
    --wasm-module-path ./ic-icrc1-archive-u256.wasm.gz \
    --wasm-module-sha256 5bd1f69540bd48493018e13bb5ad25aba75d59403ced1d5958bf718147228d31 \
    --arg args.bin \
    --summary-file ./cketh_archive_upgrade_2024_08_19.md
```

File: rs/ethereum/cketh/mainnet/cketh_index_upgrade_2024_08_19.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
+# Proposal to upgrade the ckETH index canister
```
Comment (gregorydemay) [inline rs/ethereum/cketh/mainnet/cketh_index_upgrade_2024_08_19.md:1]:
```
~/bin/ic-admin \                                                                                                                                                                                                                                                    
    --use-hsm \
    --key-id 01 \
    --slot 0 \
    --pin $(cat ~/.hsm-pin) \
    --nns-url "https://ic0.app" \
    propose-to-change-nns-canister \
    --proposer 17212304975669116357 \
    --canister-id s3zol-vqaaa-aaaar-qacpa-cai \
    --mode upgrade \
    --wasm-module-path ./ic-icrc1-index-ng-u256.wasm.gz \
    --wasm-module-sha256 eb3096906bf9a43996d2ca9ca9bfec333a402612f132876c8ed1b01b9844112a \
    --arg args.bin \
    --summary-file ./cketh_index_upgrade_2024_08_19.md
```

File: rs/ethereum/cketh/mainnet/cketh_ledger_upgrade_2024_08_19.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
+# Proposal to upgrade the ckETH ledger canister
```
Comment (gregorydemay) [inline rs/ethereum/cketh/mainnet/cketh_ledger_upgrade_2024_08_19.md:1]:
```
~/bin/ic-admin \                                                                                                                                                                                                                                                    
    --use-hsm \
    --key-id 01 \
    --slot 0 \
    --pin $(cat ~/.hsm-pin) \
    --nns-url "https://ic0.app" \
    propose-to-change-nns-canister \
    --proposer 17212304975669116357 \
    --canister-id ss2fx-dyaaa-aaaar-qacoq-cai \
    --mode upgrade \
    --wasm-module-path ./ic-icrc1-ledger-u256.wasm.gz \
    --wasm-module-sha256 8457289d3b3179aa83977ea21bfa2fc85e402e1f64101ecb56a4b963ed33a1e6 \
    --arg args.bin \
    --summary-file ./cketh_ledger_upgrade_2024_08_19.md
```

File: rs/ethereum/cketh/mainnet/cketh_archive_upgrade_2024_08_19.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+Git hash: `3d0b3f10417fc6708e8b5d844a0bac5e86f3e17d`
+
+New compressed Wasm hash: `5bd1f69540bd48493018e13bb5ad25aba75d59403ced1d5958bf718147228d31`
```
Comment (ninegua) [inline rs/ethereum/cketh/mainnet/cketh_archive_upgrade_2024_08_19.md:5]: Hash confirmed.

File: rs/ethereum/cketh/mainnet/cketh_index_upgrade_2024_08_19.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+Git hash: `3d0b3f10417fc6708e8b5d844a0bac5e86f3e17d`
+
+New compressed Wasm hash: `eb3096906bf9a43996d2ca9ca9bfec333a402612f132876c8ed1b01b9844112a`
```
Comment (ninegua) [inline rs/ethereum/cketh/mainnet/cketh_index_upgrade_2024_08_19.md:5]: Hash confirmed.

File: rs/ethereum/cketh/mainnet/cketh_ledger_upgrade_2024_08_19.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+
+Git hash: `3d0b3f10417fc6708e8b5d844a0bac5e86f3e17d`
+
+New compressed Wasm hash: `8457289d3b3179aa83977ea21bfa2fc85e402e1f64101ecb56a4b963ed33a1e6`
```
Comment (ninegua) [inline rs/ethereum/cketh/mainnet/cketh_ledger_upgrade_2024_08_19.md:5]: Hash confirmed.

File: rs/ethereum/cketh/mainnet/cketh_archive_upgrade_2024_08_19.md (hunk 1 above)
Comment (ninegua) [reply rs/ethereum/cketh/mainnet/cketh_archive_upgrade_2024_08_19.md:23]: Checked.

File: rs/ethereum/cketh/mainnet/cketh_index_upgrade_2024_08_19.md (hunk 2 above)
Comment (ninegua) [reply rs/ethereum/cketh/mainnet/cketh_index_upgrade_2024_08_19.md:23]: Checked.

File: rs/ethereum/cketh/mainnet/cketh_ledger_upgrade_2024_08_19.md (hunk 3 above)
Comment (ninegua) [reply rs/ethereum/cketh/mainnet/cketh_ledger_upgrade_2024_08_19.md:23]: checked.

**Review by ninegua — APPROVED** [review]

File: rs/ethereum/cketh/mainnet/cketh_archive_upgrade_2024_08_19.md (hunk 1 above)
Comment (THLO) [reply rs/ethereum/cketh/mainnet/cketh_archive_upgrade_2024_08_19.md:23]: Verified!

File: rs/ethereum/cketh/mainnet/cketh_index_upgrade_2024_08_19.md (hunk 2 above)
Comment (THLO) [reply rs/ethereum/cketh/mainnet/cketh_index_upgrade_2024_08_19.md:23]: Verified!

File: rs/ethereum/cketh/mainnet/cketh_ledger_upgrade_2024_08_19.md (hunk 9 above)
Comment (THLO) [reply rs/ethereum/cketh/mainnet/cketh_ledger_upgrade_2024_08_19.md:5]: Hash confirmed!

**Review by THLO — APPROVED** [review]

**Pull request merged**